pkg/xds/envoy/clusters/v3/client_side_tls_configurer.go

```diff
@@ -23,13 +21,18 @@ var _ ClusterConfigurer = &ClientSideTLSConfigurer{}
 func (c *ClientSideTLSConfigurer) Configure(cluster *envoy_cluster.Cluster) error {
 	for _, ep := range c.Endpoints {
 		if ep.ExternalService.TLSEnabled {
+			sni := ep.Target
```

Comment:
SNI can't be an IP address literal ... are we able to make that guarantee here?

Reply:
Good point. Technically SNI can be anything (we are using our format of SNI for cross zone communication), but at least for autogenerated SNI this should only take into account hostname